subprojects/resources/src/test/groovy/org/gradle/internal/resource/UriTextResourceTest.groovy

```diff
@@ -188,15 +183,15 @@ class UriTextResourceTest extends Specification {
     }
 
     def hasNoContentWhenUsingFileUriAndFileDoesNotExist() {
-        UriTextResource resource = new UriTextResource('<display-name>', fileUri, resolver)
-        assertFalse(resource.exists)
-        assertNull(resource.file)
-        try {
-            resource.text
-            fail()
-        } catch (MissingResourceException e) {
-            assertThat(e.message, equalTo("Could not read <display-name> '$file' as it does not exist." as String))
-        }
+        when:
+        UriTextResource resource = UriTextResource.from('<display-name>', file, resolver)
+
+        then:
+        resource.exists
+        resource.file == null
+        resource.location.file != null
+        file == resource.getLocation().getFile()
```

Comment:
❓ Why not joining 192 and 193 into a single assertion `resource.location.file == file`?

Reply:
Because it would pass if both `file` and `resource.getLocation().getFile()` were null. Alternatively, I could have added an assertion for `file != null` and that would have given me enough assurance.

Reply:
I don't see how `file` can be null.

Reply:
It is more like whenever I see an assertion that could pass in two different situations (without inspecting other code), I feel an impulse to add another assertion that makes it obvious only one situation would pass the test. Also, I like that it clearly spells: "not only this is not supposed to be null, but actually it should be equal to the object held by this field over here".

Reply:
Applied your suggestion, btw